pkg/compiler/transformers/k8s/workflow.go

```diff
@@ -171,6 +171,7 @@ func BuildFlyteWorkflow(wfClosure *core.CompiledWorkflowClosure, inputs *core.Li
 	if err != nil {
 		errs.Collect(errors.NewWorkflowBuildError(err))
 	}
+	obj.ObjectMeta.Labels[WorkflowIDLabel] = primarySpec.ID
```

Comment:
There is a limit on the length of label fields, beyond which the creation of the CRD will fail... we should probably add code to guard against that.

Reply:
fixed, it was more to check if this is the right place to inject this.